### ISSUE TITLE
Retranscoding videos leads to duplicate records in open edX Studio

### ISSUE THREAD

**Issue by pdpinch** [mitodl/odl-video-service#1081]
### Steps to Reproduce

1. Create a collection and connect it to an open edX course
2. Upload a video
3. Retranscode the video 

### Expected Behavior

- There should only be one video in open edX Studio, just as there is only one video in OVS. 

### Actual Behavior

- It appears that retranscoding creates a new video record in open edX. Sometimes up to 4. See https://studio.mitxonline.mit.edu/videos/course-v1:MITxT+8.03x+3T2023 and the attached CSV, [8.03x_video_urls.csv](https://github.com/mitodl/odl-video-service/files/12152373/8.03x_video_urls.csv)

### Related Issues

- This was initially noticed when someone tried to upload 65 videos at once, and it led to #1080 and multiple videos failing to transcode. I manually set them to retranscode and then only later realized that there were multiple duplicates in open edX Studio.

**Comment by collinpreston** [comment]
So I believe this issue is caused when a Video with a status of `Complete` and in a Collection with a course ID is saved more than twice.  This scenario can occur if a Video is scheduled for retranscoding multiple times.

To resolve this issue we can record the edX video ID after creating the edX video record.  When a Video with a status of `Complete` in a collection with a course ID is saved, we can make an update to the existing edX video record using the edX video ID.

In order to obtain and store all of the edX video IDs for our existing Videos in OVS, we can create a Django management script to call an edX API endpoint (https://courses.edx.org/api-docs/#/val/val_v0_videos_list) which will list all of the video records in edX.  The script can then match the Video's title and associated Collection's course ID in OVS with the `client_video_id` and `courses` in the API response in order to determine the edX video ID for each Video in OVS.

**Comment by pdpinch** [comment]
A management command makes sense to me for backfilling edX video ids, but is there a way to get the video id immediately after posting the video to edX Studio?

**Comment by collinpreston** [comment]
> A management command makes sense to me for backfilling edX video ids, but is there a way to get the video id immediately after posting the video to edX Studio?

Yes, when OVS makes the POST API call to edX, OVS defines the edX video ID.  If the POST call is successful, then the edX video ID used to make the call should be saved with the OVS Video record.
